lib/ff/ruby/server/sdk/api/inner_client.rb

```diff
@@ -122,7 +122,7 @@ def on_auth_success
   def on_auth_failed
     SdkCodes::warn_auth_failed_srv_defaults @config.logger
     @initialized = true
-    @condition.signal
+    @condition.broadcast
```

Comment:
This condition wasn't synchronised before your changing it to a broadcast, but just noticing we'll need to syncronise it now